hooks/useApproval.ts

```diff
@@ -0,0 +1,67 @@
+import { ethers, Signer } from "ethers"
+import { useEffect, useState } from "react"
+import { erc20ABI, useAccount, useConnect, useContract, useNetwork, useProvider, useSigner, useWebSocketProvider } from "wagmi"
+
+const APPROVE_CALLBACK_STATUS = {
```

Comment:
Good! I'd like to unify the entire system to return callback constants in this format.